concep_Timeword/conceptual.md

```diff
@@ -1,30 +1,52 @@
-### Conceptual Exercise
+## Conceptual Exercise
 
 Answer the following questions below:
 
-- What is a JWT?
+#### What is a JWT?
+* JSON Web Token - a compact URL-safe means of representing claims to be transferred between two parties.
+In other words, JWTs are credentials, which can grant access to resources.
 
-- What is the signature portion of the JWT?  What does it do?
+#### What is the signature portion of the JWT?  What does it do?
+* The signature is used to verify that the sender of the JWT is who it says it is and to ensure that the message wasn't changed along the way. To create the signature, the Base64-encoded header and payload are taken, along with a secret, and signed with the algorithm specified in the header.
 
-- If a JWT is intercepted, can the attacker see what's inside the payload?
+#### If a JWT is intercepted, can the attacker see what's inside the payload?
+* Yes
 
-- How can you implement authentication with a JWT?  Describe how it works at a high level.
+#### How can you implement authentication with a JWT?  Describe how it works at a high level.
+* After you log a user in on your server, you can generate and respond with a JWT that includes info like the username or a property like "isAuthenticated". You would then include this token in any requests back to the server from the client. The server can then respond if the token is verified or deny the request if it fails verification.
 
-- Compare and contrast unit, integration and end-to-end tests.
+#### Compare and contrast unit, integration and end-to-end tests.
+* Unit tests are small, isolated tests that test only a specific function or components of code  
+* Integration tests are for testing that two or more units are able to work together. Usually done after unit testing.
+* Testing an application's workflow from beginning to end.  Often difficult to write and keep up to date as apps get modified.
 
-- What is a mock? What are some things you would mock?
+#### What is a mock? What are some things you would mock?
+* Mock objects simulate the behavior of complex real objects - and thus are useful when a real object is impractical/impossible to unit test.  According to wikipedia, mocks are useful when an object has any of the following characteristics:

```

Comment:
Mocks are usually used specifically for unit tests so you don't have to depend on / so you can control dependent functions/objects behavior. It's the standard way to isolate a single "unit"'s behavior even if it depends on other "units".